modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java

```diff
@@ -1821,7 +1822,24 @@ protected void applyFts() {
 
             if ((applyTo != null) && ListComponent.class.isAssignableFrom(applyTo.getClass())) {
                 if (clientConfig.getGenericFilterFtsTableTooltipsEnabled()) {
-                    filterHelper.initTableFtsTooltips((ListComponent) applyTo, adapter.getMetaClass(), searchTerm);
+                    filterHelper.initTableFtsTooltips(
+                            (ListComponent) applyTo,
+                            (listComponent, entityId) -> {
```

Comment:
Why are we using bi-function here? We have a reference to ListComponent, so we are able to extract metaclasses without additional bi-function parameter from the ListComponent.

Reply:
I thought about. But first, `FilterDelegateImpl` incapsulates work with `DataSource` or `DataLoader` in `FilterDelegate#Adapter`. And second, when a look to `WebFilterHelper` class, I admit that it designed in some another abstraction layer and hasn't to depend on underlying classes, working with Data (`DataSource` or `DataLoader`). Thus, I used bi-function to let `FilterDelegateImpl` work with `DataSource` or `DataLoader`